core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -414,7 +414,7 @@ struct HIPParallelLaunchKernelInvoker<DriverType, LaunchBounds,
 
     // Copy functor (synchronously) to staging buffer in pinned host memory
     unsigned long *staging = hip_instance->constantMemHostStaging;
-    std::memcpy(staging, &driver, sizeof(DriverType));
+    std::memcpy((void *)staging, (void *)&driver, sizeof(DriverType));
```

Comment:
We should use C++ casts and be const correct (as the C-style cast on `&driver` isn't), as in
```c++
std::memcpy(static_cast<void*>(staging), static_cast<const void*>(&driver), sizeof(driver));
```